docs/animation/armature.md

```diff
@@ -0,0 +1,183 @@
+Armature Files
+==============
+
+Armature Files define joints and clips for animating a model.
+
+File Structure
+----------------
+
+An example armature file, taken from the forge debug mod (requires #3875)
+```json
+{
+  "joints": {
+    "stick": {"2": [1.0]},
+    "cube": {"3": [1.0]}
+  },
+  "clips": {
+    "default": {
+      "loop": true,
+      "joint_clips": {
+        "stick": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          }
+        ],
+        "cube": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          },
+          {
+            "variable": "axis_z",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 1 ]
+          },
+          {
+            "variable": "origin_x",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.15625 ]
+          },
+          {
+            "variable": "origin_y",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.40625 ]
+          },
+          {
+            "variable": "angle",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 120, 240, 0, 120, 240]
+          }
+        ]
+      },
+      "events": {}
+    }
+  }
+}
+
+```
+
+The file is organized in two sections, joints and clips.
+
+Joints
+--------
+Each joint defines a connection to the model file. A joint can animate any number of elements in the model file, but they are all animated with the same input value and transforms
```

Comment:
"a connection to the model file" feels a bit ambiguous. How about

`A joint defines an object that animates together as a single unit in the animation together. For example, in the vanilla json model format, multiple elements can belong to the same joint.`

Reply:
sound good to me